components/bytes/src/lib.rs

```diff
@@ -245,6 +251,20 @@ impl<'a, B: MemBufMut> Write for Writer<'a, B> {
     }
 }
 
+impl Write for &mut dyn MemBufMut {
+    fn write(&mut self, src: &[u8]) -> io::Result<usize> {
+        self.write_slice(src).map_err(|e| match &e {
+            Error::UnexpectedEof { .. } => io::Error::new(io::ErrorKind::UnexpectedEof, e),
+            Error::WouldOverflow { .. } => io::Error::new(io::ErrorKind::WriteZero, e),
+        })?;
+        Ok(src.len())
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        Ok(())
+    }
+}
+
```

Comment:
`&mut dyn MemBufMut` can not be wrapped to `Writer`, because the size of `dyn MemBufMut` can not know at compile time.
So I directly implement `Write` for `&mut dyn MemBufMut`.